src/sw.ts

```diff
@@ -101,7 +105,46 @@ self.addEventListener('fetch', (event) => {
     // intercept and do our own stuff...
     event.respondWith(fetchHandler({ path: url.pathname, request }))
   } else if (isSubdomainRequest(event)) {
-    event.respondWith(fetchHandler({ path: url.pathname, request }))
+    const isMutable = protocol === 'ipns'
+    const cacheKey = `${event.request.url}-${event.request.headers.get('Accept') ?? ''}`
+    const cacheName = isMutable ? MUTABLE_CACHE : IMMUTABLE_CACHE
+    log('helia-sw: cache name: %s | key: %s', cacheName, cacheKey)
+
+    event.respondWith((async () => {
```

Comment:
we should move this into a separate function

Reply:
Moved it into a separate function.